upgrader.go

```diff
@@ -71,17 +72,17 @@ func (u *Upgrader) upgrade(ctx context.Context, t transport.Transport, maconn ma
 	}
 
 	var conn net.Conn = maconn
-	if u.Protector != nil {
-		pconn, err := u.Protector.Protect(conn)
+	if u.PSK != nil {
+		pconn, err := pnet.NewProtectedConn(u.PSK, conn)
```

Comment:
future: We may want to eventually expose this as a "Protect" function so transports can choose which pieces of the upgrader they want to use.